### PR TITLE
[#PE-632] Added Search function to cgn operator search api

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### List of Changes
- updated openapi specs
- generated new client and models
- added new path for cgn operator search api
- added tests

#### Motivation and Context
- we are improving operator search functionalities

#### How Has This Been Tested?
- yarn build
- yarn lint
- yarn test

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api_cgn_operator_search.yaml [hunk 1]
```diff
@@ -29,6 +29,28 @@ paths:
...
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/SearchRequest'
+      responses:
+        "200":
+          description: List of merchants/discounts for the given query parameters
+          schema:
+            $ref: "#/definitions/SearchResult"
+        "500":
```
Comment (gquadrati) [inline api_cgn_operator_search.yaml:49]: I'd add at lease the following response codes:

```typescript
        "400":
          description: Bad request
        "401":
          description: Token null or invalid.
```

File: src/controllers/cgnOperatorSearchController.ts [hunk 2]
```diff
@@ -80,6 +82,24 @@ export default class CgnOperatorSearchController {
...
 
+  /**
+   * Search CGN merchants/discounts that matches with search criteria
+   */
+  public readonly search = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorValidation
+    | IResponseErrorInternal
+    | IResponseErrorNotFound
```
Comment (gquadrati) [inline src/controllers/cgnOperatorSearchController.ts:93]: 404 is not listed as possible status code

**Review by gquadrati — APPROVED** [review]
LGTM, after fixing the comments.

File: api_cgn_operator_search.yaml (hunk 1 above)
Comment (michaeldisaro) [reply api_cgn_operator_search.yaml:49]: fixed in https://github.com/pagopa/io-backend/pull/1161/commits/ac49ea4395eae8b562fcb467639c955239f0996e

added just 400 and 401, 404 is not in this endpoint but is just present in the common type.

File: src/controllers/cgnOperatorSearchController.ts (hunk 2 above)
Comment (michaeldisaro) [reply src/controllers/cgnOperatorSearchController.ts:93]: fixed in https://github.com/pagopa/io-backend/pull/1161/commits/ac49ea4395eae8b562fcb467639c955239f0996e

**Pull request merged**